### PR TITLE
Support for loading, unloading and listing Redis Modules

### PR DESCRIPTION
**Pull request by RoeyPrat**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `$ tox` pass with this change (including linting)?
- [x] Does travis tests pass with this change (enable it first in your forked repo and wait for the travis build to finish)?
- [ ] Is the new or changed code fully tested?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?

_NOTE: these things are not required to open a PR and can be done
afterwards / while the PR is open._

### Description of change
This is a continuation of @vamc19 's PR: https://github.com/andymccurdy/redis-py/pull/808

### CODE REVIEW

**Comment by RoeyPrat** [comment]
tested manually with this script:
```
import urllib.request
import redis
import os
import stat

urllib.request.urlretrieve(
    "https://github.com/RedisJSON/RedisJSON/releases/download/v1.0.4/rejson.so",
    "rejson.so")
st = os.stat("/home/vagrant/redis-py/rejson.so")
os.chmod("/home/vagrant/redis-py/rejson.so", st.st_mode | stat.S_IEXEC)
r = redis.Redis()
r.module_load("/home/vagrant/redis-py/rejson.so")
print(r.module_list())
r.module_unload('ReJSON')
```
the unload action fails because rejson has commands, and you can't unload a module with commands.

File: redis/exceptions.py [hunk 1]
```diff
@@ -57,6 +57,10 @@ class NoPermissionError(ResponseError):
     pass
 
 
+class ModuleError(RedisError):
```
Comment (andymccurdy) [inline redis/exceptions.py:60]: `ModuleError` should inherit from `ResponseError` rather than `RedisError`.

Comment (RoeyPrat) [reply redis/exceptions.py:60]: OK

**Comment by RoeyPrat** [comment]
This code is not fully tested, and this fails the codecov checks.
The test idea we've had requires compiling a Redis module, which will make the testing a bit too complex for what this is worth.
Perhaps after we move to docker test env we can revisit the test.
For now I'm merging as-is @andymccurdy

**Pull request merged**